In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
pizza = pd.read_csv("https://raw.githubusercontent.com/rajeevratan84/data-analyst-bootcamp/master/8358_1.csv")
num_rows, num_cols = pizza.shape
#define df.shape with the number of rows and columns
print(f"Table contains {num_rows} rows and {num_cols} columns")

Table contains 3510 rows and 21 columns


In [18]:
pizza.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3510 entries, 0 to 3509
Data columns (total 21 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   id                  3510 non-null   object 
 1   address             3510 non-null   object 
 2   categories          3510 non-null   object 
 3   city                3510 non-null   object 
 4   country             3510 non-null   object 
 5   keys                3510 non-null   object 
 6   latitude            3510 non-null   float64
 7   longitude           3510 non-null   float64
 8   menuPageURL         610 non-null    object 
 9   menus.amountMax     2948 non-null   float64
 10  menus.amountMin     2948 non-null   float64
 11  menus.currency      2951 non-null   object 
 12  menus.dateSeen      3510 non-null   object 
 13  menus.description   1526 non-null   object 
 14  menus.name          3510 non-null   object 
 15  name                3510 non-null   object 
 16  postal

In [19]:
pizza['menus.amountMax'].isnull().values.any()
pizza['menus.amountMax'].isnull().sum()

562

In [20]:
pizzadrop=pizza.dropna(subset=['menus.amountMax', 'menus.amountMin'])
num_rows, num_cols = pizzadrop.shape
print(f"Table contains {num_rows} rows and {num_cols} columns")

Table contains 2948 rows and 21 columns


In [21]:
maxmean=pizzadrop['menus.amountMax'].mean()
minmean=pizzadrop['menus.amountMin'].mean()
print(f"The mean of pizza max price is {maxmean}, and the mean of pizza min price is {minmean}")

The mean of pizza max price is 12.479185888738241, and the mean of pizza min price is 11.427849389416831


Since we do not want to waste the data which include Nah in columns (menus.amountMax,menus.amountMin), we replace the nah columns with their max mean(12.4792) and min mean(11.4278) by imputation method.

In [33]:
import numpy as np
from sklearn.impute import SimpleImputer

mean_imputer = SimpleImputer(strategy='mean')
pizza['menus.amountMax'] = mean_imputer.fit_transform(pizza['menus.amountMax'].values.reshape(-1,1))
pizza['menus.amountMin'] = mean_imputer.fit_transform(pizza['menus.amountMin'].values.reshape(-1,1))


In [34]:
pizza['menus.amountMax'].isnull().sum()


0

In [35]:
pizza['menus.amountMin'].isnull().sum()

0

In [36]:
maxmean=pizza['menus.amountMax'].mean()
minmean=pizza['menus.amountMin'].mean()
print(f"The mean of pizza max price is {maxmean}, and the mean of pizza min price is {minmean}")

The mean of pizza max price is 12.479185888738076, and the mean of pizza min price is 11.42784938941656


After the imputation, the means of two columns still remind the same.

In [39]:
pizza['average_price'] = (pizza['menus.amountMax'] + pizza['menus.amountMin']) / 2
pizza.head()

id                                  address  \
0  AVwc_6KEIN2L1WUfrKAH  Cascade Village Mall Across From Target   
1  AVwc_6KEIN2L1WUfrKAH  Cascade Village Mall Across From Target   
2  AVwc_6qRByjofQCxkcxw                     148 S Barrington Ave   
3  AVwc_6qRByjofQCxkcxw                     148 S Barrington Ave   
4  AVwc_6qRByjofQCxkcxw                     148 S Barrington Ave   

                       categories         city country  \
0                     Pizza Place         Bend      US   
1                     Pizza Place         Bend      US   
2  American Restaurant,Bar,Bakery  Los Angeles      US   
3  American Restaurant,Bar,Bakery  Los Angeles      US   
4  American Restaurant,Bar,Bakery  Los Angeles      US   

                                                keys   latitude   longitude  \
0  us/or/bend/cascadevillagemallacrossfromtarget/...  44.102665 -121.300797   
1  us/or/bend/cascadevillagemallacrossfromtarget/...  44.102665 -121.300797   
2  us/brentwood/losangeles/148sbarringtonave/-151...  34.064563 -118.469017   
3  us/brentwood/losangeles/148sbarringtonave/-151...  34.064563 -118.469017   
4  us/brentwood/losangeles/148sbarringtonave/-151...  34.064563 -118.469017   

  menuPageURL  menus.amountMax  ...        menus.dateSeen  \
0         NaN            22.50  ...  2016-03-31T02:19:42Z   
1         NaN            18.95  ...  2016-03-31T02:19:42Z   
2         NaN            12.00  ...  2015-10-23T03:57:26Z   
3         NaN            13.00  ...  2015-10-23T03:57:26Z   
4         NaN            13.00  ...  2015-10-23T03:57:26Z   

                  menus.description         menus.name                   name  \
0                               NaN       Bianca Pizza  Little Pizza Paradise   
1                               NaN       Cheese Pizza  Little Pizza Paradise   
2                               NaN  Pizza, Margherita          The Brentwood   
3                               NaN    Pizza, Mushroom          The Brentwood   
4  Olives, onions, capers, tomatoes  Pizza, Puttenesca          The Brentwood   

  postalCode priceRangeCurrency priceRangeMin priceRangeMax   province  \
0      97701                NaN           NaN           NaN         OR   
1      97701                NaN           NaN           NaN         OR   
2      90049                USD          50.0          55.0  Brentwood   
3      90049                USD          50.0          55.0  Brentwood   
4      90049                USD          50.0          55.0  Brentwood   

   average_price  
0          19.00  
1          18.95  
2          12.00  
3          13.00  
4          13.00  

[5 rows x 22 columns]

In [42]:
pizza.to_csv("/Users/georgeli/Desktop/stat3011/Project 2/newpizzadata.csv")